ci-operator/step-registry/upi/conf/vsphere/dns/upi-conf-vsphere-dns-commands.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if command -v pwsh &> /dev/null
+then
+  echo "vmc-ci.devcluster.openshift.com" > "${SHARED_DIR}"/basedomain.txt
+
+  cluster_name=${NAMESPACE}-${UNIQUE_HASH}
+  base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
+  cluster_domain="${cluster_name}.${base_domain}"
+
+  export AWS_DEFAULT_REGION=us-west-2  # TODO: Derive this?
+  export AWS_SHARED_CREDENTIALS_FILE=/var/run/vault/vsphere/.awscred
+  export AWS_MAX_ATTEMPTS=50
+  export AWS_RETRY_MODE=adaptive
+  export HOME=/tmp
+
+  if ! command -v aws &> /dev/null
+  then
+      echo "$(date -u --rfc-3339=seconds) - Install AWS cli..."
+      export PATH="${HOME}/.local/bin:${PATH}"
+      if command -v pip3 &> /dev/null
+      then
+          pip3 install --user awscli
+      else
+          if [ "$(python -c 'import sys;print(sys.version_info.major)')" -eq 2 ]
+          then
+            easy_install --user 'pip<21'
+            pip install --user awscli
+          else
+            echo "$(date -u --rfc-3339=seconds) - No pip available exiting..."
+            exit 1
+          fi
+      fi
+  fi
+
+  # Load array created in setup-vips:
+  # 0: API
+  # 1: Ingress
+  declare -a vips
+  mapfile -t vips < "${SHARED_DIR}"/vips.txt
+
+  hosted_zone_id="$(aws route53 list-hosted-zones-by-name \
+              --dns-name "${base_domain}" \
+              --query "HostedZones[? Config.PrivateZone != \`true\` && Name == \`${base_domain}.\`].Id" \
+              --output text)"
+  echo "${hosted_zone_id}" > "${SHARED_DIR}/hosted-zone.txt"
+
+  if [ "${JOB_NAME_SAFE}" = "launch" ]; then
+    # setup DNS records for clusterbot to point to the IBM VIP
+    api_dns_target='"TTL": 60,
+      "ResourceRecords": [{"Value": "'${vips[0]}'"}, {"Value": "169.48.190.20"}]'
+    apps_dns_target='"TTL": 60,
+      "ResourceRecords": [{"Value": "169.48.190.20"}]'
+  else
+    # Configure DNS direct to respective VIP
+    api_dns_target='"TTL": 60,
+          "ResourceRecords": [{"Value": "'${vips[0]}'"}]'
+    apps_dns_target='"TTL": 60,
+          "ResourceRecords": [{"Value": "'${vips[1]}'"}]'
+  fi
+
+  # api-int record is needed just for Windows nodes
+  # TODO: Remove the api-int entry in future
+  echo "Creating DNS records..."
+  cat > "${SHARED_DIR}"/dns-create.json <<EOF
+{
+"Comment": "Create public OpenShift DNS records for VSphere IPI CI install",
```

Comment:
```suggestion
"Comment": "Create public OpenShift DNS records for VSphere UPI CI install",
```

Reply:
Thanks.  I'll fix that.  This was a copy of IPI due to needing to limit when the logic is executed.  As for all other suggestions, the logic was harvested and I'll fix as much as you have suggest.  Thanks for the feedback.